### ISSUE TITLE
Allow pip installing pynamo without botocore

### ISSUE THREAD

**Issue by mew1033** [pynamodb/PynamoDB#1166]
I often use pynamodb in AWS lambda functions. In the python lambda runtime environment, botocore is already available. To package and upload my functions, I typically use AWS SAM which supports pip installing packages as part of its deploy process. Installing pynamodb via pip brings in botocore as a dependency (as is expected), which causes the entire package to be larger than 10 MB. I'm wondering if it's possible for you to provide some other way to pip install pynamodb that _doesn't_ bring in the botocore library since it's already available in the AWS lambda runtime environment.

Thanks!

**Comment by ikonst** [comment]
If it's available, why would pip install it?

**Comment by mew1033** [comment]
> If it's available, why would pip install it?

When SAM installs packages via pip, it simply takes the requirements.txt file and runs through it, installing them alongside your code. It has no knowledge of the eventual runtime the code will be running in. In this case, the AWS Lambda Python runtime, which has botocore (and multiple others) installed.

**Comment by ikonst** [comment]
Typically pip installs packages into the environment where your code will be running. If that environment already has a package (e.g. botocore) installed, then pip can be aware of that.

Are you saying SAM installs them into some environment without botocore, then those packages are somehow re-homed into a different environment that has botocore?

**Comment by mew1033** [comment]
> Are you saying SAM installs them into some environment without botocore, then those packages are somehow re-homed into a different environment that has botocore?

Correct. SAM runs on your local machine, executes pip (I actually think it's technically pip download, not install, but I'm not 100% sure) and puts all the packages it gathered into a folder with your source code. It then packages all that up as a zip file and uploads it to AWS Lmabda where it is extracted in a new execution environment. This new environment has a few small modifications from a standard python installation. One of those modifications is that there are some packages already installed.

AWS says that they only officially have two extra packages: boto3 and botocore. (See https://docs.aws.amazon.com/lambda/latest/dg/lambda-runtimes.html) But there are other packages available that have been brought in as dependencies of boto3 and botocore.
Regardless, botocore is definitely there and it works just fine to use it. I manually created a lambda function zip file with pynamodb and typing-extensions, but _not_ botocore. It works just fine.

**Comment by ikonst** [comment]
A trivial way that comes to mind is to have pip run with `--no-deps` — should be possible to set it for the specific `pynamodb` package in requirements3.txt.
